typescript/packages/subsurface-viewer/src/views/viewport.ts

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+
+/**
+ * Viewport type.
+ */
+export interface ViewportType {
+    /**
+     * Viewport id
+     */
+    id: string;
+
+    /**
+     * Viewport name
+     */
+    name?: string;
+
+    /**
+     * If true, displays map in 3D view, default is 2D view (false)
+     */
+    show3D?: boolean;
+
+    /**
+     * Layers to be displayed on viewport
+     */
+    layerIds?: string[];
+
+    target?: [number, number];
+    zoom?: number;
+    rotationX?: number;
+    rotationOrbit?: number;
+
+    /** A vertical scale factor, used to scale items in the viewport vertically */
+    verticalScale?: number;
+
+    isSync?: boolean;
+}
+
+export const useVerticalScale = (viewport: ViewportType[] | undefined) => {
```

Comment:
Please, rename `viewport` to `viewports`.

So here we use the first found verticalScale from all the viewports, right ?

Reply:
> Please, rename `viewport` to `viewports`.

Done.
> 
> So here we use the first found verticalScale from all the viewports, right ?

Yes, will make an attempt to make `verticalScale` viewport dependent, like in 2D, in a subsequent PR, and then we can apply a `verticalScale` respectively to each viewport.